packages/grafana-sql/src/components/visual-query-builder/SelectCustomFunctionParameters.tsx

```diff
@@ -0,0 +1,137 @@
+  onParameterChange: (index: number) => (value?: string) => void;
+  currentColumnIndex: number;
+}
+
+export function SelectCustomFunctionParameters({
+  columns,
+  query,
+  onSqlChange,
+  onParameterChange,
+  currentColumnIndex,
+}: Props) {
+  const styles = useStyles2(getStyles);
+  const macroOrFunction = query.sql?.columns?.[currentColumnIndex];
+
+  const addParameter = useCallback(
+    (index: number) => () => {
+      const item = query.sql?.columns?.[index];
+      if (!item) {
+        return;
+      }
+
+      item.parameters = item.parameters
+        ? [...item.parameters, { type: QueryEditorExpressionType.FunctionParameter, name: '' }]
+        : [];
+
+      const newSql: SQLExpression = {
+        ...query.sql,
+        columns: query.sql?.columns?.map((c, i) => (i === index ? item : c)),
+      };
+
+      onSqlChange(newSql);
+    },
+    [onSqlChange, query.sql]
+  );
+
+  const removeParameter = useCallback(
+    (columnIndex: number, index: number) => () => {
+      const item = query.sql?.columns?.[columnIndex];
+      if (!item?.parameters) {
+        return;
+      }
+      item.parameters = item.parameters?.filter((_, i) => i !== index);
+
+      const newSql: SQLExpression = {
+        ...query.sql,
+        columns: query.sql?.columns?.map((c, i) => (i === columnIndex ? item : c)),
+      };
+
+      onSqlChange(newSql);
+    },
+    [onSqlChange, query.sql]
+  );
+
+  function renderParameters(columnIndex: number) {
+    if (!macroOrFunction?.parameters || macroOrFunction.parameters.length <= 1) {
+      return null;
+    }
+
+    const paramComponents = macroOrFunction.parameters.map((param, index) => {
+      // Skip the first parameter as it is the column name
+      if (index === 0) {
+        return null;
+      }
+
+      return (
+        <Stack key={index} gap={2}>
+          <InlineLabel className={styles.label}>,</InlineLabel>
+          <Input
+            onChange={(e) => onParameterChange(index)(e.currentTarget.value)}
+            value={param.name}
+            aria-label={`Parameter ${index} for column ${columnIndex}`}
+            data-testid={selectors.components.SQLQueryEditor.selectInputParameter}
+            addonAfter={
+              <Button
+                title="Remove parameter"
+                type="button"
+                icon="times"
+                variant="secondary"
+                size="md"
+                onClick={removeParameter(columnIndex, index)}
+              />
+            }
+          />
+        </Stack>
+      );
+    });
+    return paramComponents;
+  }
+
+  return (
+    <>
+      <InlineLabel className={styles.label}>(</InlineLabel>
+      <SelectColumn
+        columns={columns}
+        onParameterChange={(s) => onParameterChange(0)(s)}
+        value={getColumnValue(macroOrFunction?.parameters?.[0])}
+      />
+      {renderParameters(currentColumnIndex)}
+      <Button
+        type="button"
+        onClick={addParameter(currentColumnIndex)}
```

Comment:
personal nit, I got here a little bit confused why is `addParameter` and also `removeParameter` invoked, but then I noticed that they both are:
```
  const removeParameter = useCallback(
    (columnIndex: number, index: number) => () => {
```

Any reason why not just 
```
  const removeParameter = useCallback(
    (columnIndex: number, index: number) => {

		onClick={() => addParameter(currentColumnIndex)}
```

Feel free to ignore, I was literally just curious.

Reply:
I think this just a matter of taste how you write this. I can change it so it won't be confusing.

Reply:
No need to change, I was just curious!